nautobot/core/api/renderers.py

```diff
@@ -24,3 +36,168 @@ class NautobotJSONRenderer(JSONRenderer):
     """
 
     encoder_class = NautobotKombuJSONEncoder
+
+
+class NautobotCSVRenderer(BaseRenderer):
+    """
+    Render to CSV format.
+
+    Loosely inspired by https://github.com/mjumbewu/django-rest-framework-csv/.
+    """
+
+    media_type = "text/csv"
+    format = "csv"
+    charset = "UTF-8"
+
+    def render(self, data, accepted_media_type=None, renderer_context=None):
+        """
+        Render the provided data to CSV format.
+
+        Unlike other DRF Renderers, `data` should be a dict with key "serializer_class" (the serializer for
+        this view) and either of the keys "instance" (for a single object) or "queryset" (for a list of objects).
+        """
+        if not data:
+            return ""
+
+        if not isinstance(data, dict) or "serializer_class" not in data:
+            raise ValueError(
+                "data should be a dict with the keys 'serializer_class' and either 'instance' or 'queryset'."
+            )
+
+        serializer_class = data["serializer_class"]
+        objects = data.get("queryset", [data.get("instance")])
+        if not objects:
+            return ""
+
+        headers = self.get_headers(serializer_class(objects[0], context={"request": renderer_context["request"]}))
+
+        buffer = StringIO()
+        writer = csv.writer(buffer)
+        writer.writerow(headers)
+        for obj in objects:
+            writer.writerow(
+                self.object_to_row_elements(
+                    obj,
+                    headers=headers,
+                    serializer=serializer_class(obj, context={"request": renderer_context["request"]}),
+                )
+            )
+
+        return buffer.getvalue()
+
+    @classmethod
+    def get_headers(cls, serializer):
+        """Identify the appropriate CSV headers corresponding to the given serializer."""
+        base_headers = list(serializer.fields)
+
+        # Remove specific headers that we know are irrelevant
+        for undesired_header in [
+            "computed_fields",
```

Comment:
I like the restraint of this for the first go-around but I'd like to consider opt-in ability of some of these fields (also noting #3610).

Reply:
I can probably just let `config_context` and `local_config_context_data` be serialized as JSON (ugly though CSV-escaped JSON is!), as it seems to be working well enough for other JSON fields like `User.config_data`.